### ISSUE TITLE
Can Connection#send() thrown an error?

### ISSUE THREAD

**Issue by ibc** [theturtle32/WebSocket-Node#185]
If I call `send()` or `sendUTF()` or `sendBytes()` on an already closed `Connection`, or if sending fails due to connection error, would it throw an error?

It is not documented.

**Comment by theturtle32** [comment]
No.  The behavior for this is inherited from the `Socket` implementation in Node core.  If there is an error, such as when writing to a closed socket, it will emit an `error` event on `nextTick`.  I believe earlier versions of Node did throw in some cases, so I used to normalize that behavior using `try`/`catch`, but that's not necessary anymore because Node never throws when calling `write` except when the user calls it with invalid parameters.

The main difference here is that if nobody is listening for an `error` event on the `WebSocketConnection` instance, it will silently swallow the error event instead of letting it bubble up and crash the application.  My reasoning for that was that it will actually be pretty common to run into these errors in production, and it will immediately trigger a `close` event, and there's no other handling that needs to be done by the user for these errors.  I do expose the `error` event for anyone who finds it useful in their particular situation, but if no listeners are added for the `error` event, then any `error` events emitted by the underlying `Socket` will be silently discarded.

See [io.js/net.js - writeAfterFIN](https://github.com/iojs/io.js/blob/cca8de6709ece5b4791d52a8b834da83bfb782d5/lib/net.js#L263)

See also [WebSocketConnection.prototype.handleSocketError](https://github.com/theturtle32/WebSocket-Node/blob/e0e6d580a4253aafd2eb0cc1f1577ed0ca173e15/lib/WebSocketConnection.js#L330)

**Comment by theturtle32** [comment]
Should I add a comment to the documentation to this effect?

**Comment by ibc** [comment]
Thanks for the explanation! Yes, I would find some text about this subject really useful in the API doc :)
